app/init/draft_upload_manager.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ClientResponse, ClientResponseError} from '@mattermost/react-native-network-client';
+import {AppState, AppStateStatus, DeviceEventEmitter} from 'react-native';
+
+import {updateDraftFile} from '@actions/local/draft';
+import {uploadFile} from '@actions/remote/file';
+import {Events} from '@constants';
+import {PROGRESS_TIME_TO_STORE} from '@constants/files';
+
+type FileHandler = {
+    [clientId: string]: {
+        cancel?: () => void;
+        fileInfo: FileInfo;
+        serverUrl: string;
+        channelId: string;
+        rootId: string;
+        lastTimeStored: number;
+    };
+}
+
+class DraftUploadManager {
+    private handlers: FileHandler = {};
+    private previousAppState: AppStateStatus;
+
+    constructor() {
+        this.previousAppState = AppState.currentState;
+        AppState.addEventListener('change', this.onAppStateChange);
+    }
+
+    public prepareUpload = (
+        serverUrl: string,
+        file: FileInfo,
+        channelId: string,
+        rootId: string,
+        skipBytes = 0,
+    ) => {
+        this.handlers[file.clientId!] = {
+            fileInfo: file,
+            serverUrl,
+            channelId,
+            rootId,
+            lastTimeStored: 0,
+        };
+
+        const onProgress = (progress: number) => {
+            this.handleProgress(file.clientId!, progress);
+        };
+
+        const onComplete = (response: ClientResponse) => {
+            this.handleComplete(response, file.clientId!);
+        };
+
+        const onError = (response: ClientResponseError) => {
+            this.handleError(response.message, file.clientId!);
+        };
+
+        const {error, cancel} = uploadFile(serverUrl, file, channelId, onProgress, onComplete, onError, skipBytes);
+        if (error) {
+            this.handleError(error.message, file.clientId!);
+            return;
+        }
+        this.handlers[file.clientId!].cancel = cancel;
+    };
+
+    public cancel = (clientId: string) => {
+        const {cancel} = this.handlers[clientId];
+        delete this.handlers[clientId];
+        cancel?.();
+    };
+
+    public isLoading = (clientId: string) => {
```

Comment:
If we can the clientId in this.handers, that means it is currently being uploaded?  If yes, perhaps, we can rename the method `isLoading` to `isUploading` ?